src/Features/LanguageServer/Protocol/Features/Options/CompletionOptionsStorage.cs

```diff
@@ -29,8 +29,8 @@ public static CompletionOptions GetCompletionOptions(this IGlobalOptionService o
             ForceExpandedCompletionIndexCreation = options.GetOption(ForceExpandedCompletionIndexCreation),
             UpdateImportCompletionCacheInBackground = options.GetOption(UpdateImportCompletionCacheInBackground),
             NamingStyleFallbackOptions = options.GetNamingStylePreferences(language),
-            ShowNewSnippetExperience = options.GetOption(ShowNewSnippetExperience, language),
-            SnippetCompletion = options.GetOption(ShowNewSnippetExperienceFeatureFlag)
+            ShowNewSnippetExperienceUserOption = options.GetOption(ShowNewSnippetExperience, language),
+            SnippetCompletionFeatureFlag = options.GetOption(ShowNewSnippetExperienceFeatureFlag)
```

Comment:
Is it easier just to have a single boolean property here, and this code just does the ?? to select one and then the other? That way the CompletionOptions is more a clean data representation and doesn't concern itselfe with policy. Feel free to ignore this thought since having this change in is better than no change at all.